internal/cpugroup/cpugroup.go

```diff
@@ -0,0 +1,76 @@
+package cpugroup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const NullGroupID = "00000000-0000-0000-0000-000000000000"
+
+// ErrHVStatusInvalidCPUGroupState corresponds to the internal error code for HV_STATUS_INVALID_CPU_GROUP_STATE
+var ErrHVStatusInvalidCPUGroupState = errors.New("The hypervisor could not perform the operation because the CPU group is entering or in an invalid state.")
+
+// DeleteCPUGroup deletes the cpugroup from the host
+func DeleteCPUGroup(ctx context.Context, id string) error {
+	operation := hcsschema.DeleteGroup
+	details := hcsschema.DeleteGroupOperation{
+		GroupId: id,
+	}
+
+	return modifyCPUGroupRequest(ctx, operation, details)
+}
+
+// modifyCPUGroupRequest is a helper function for making modify calls to a cpugroup
+func modifyCPUGroupRequest(ctx context.Context, operation hcsschema.CPUGroupOperation, details interface{}) error {
+	req := hcsschema.ModificationRequest{
+		PropertyType: hcsschema.PTCPUGroup,
+		Settings: &hcsschema.HostProcessorModificationRequest{
+			Operation:        operation,
+			OperationDetails: details,
+		},
+	}
+
+	return hcs.ModifyServiceSettings(ctx, req)
+}
+
+// CreateNewCPUGroupWithID creates a new cpugroup on the host with a prespecified id
+func CreateNewCPUGroupWithID(ctx context.Context, id string, logicalProcessors []uint32) error {
```

Comment:
Can this just be `CreateCPUGroup`? If we had another function that created a cpugroup with an ID that was created internally I feel this callout of WithID would be warranted but we don't currently. Also seeming as how the package is already called cpugroup, would you be opposed for creation and deletion just being `Create` and `Delete`?